.github/workflows/python.yml

```diff
@@ -197,6 +208,8 @@ jobs:
 
     lint:
         runs-on: ubuntu-latest
+        needs: get-matrices
+        if: ${{ needs.get-matrices.outputs.should-run == 'true' }}
```

Comment:
why? we don't need the matrices to test the lint, and we always run it

Reply:
Same as above. And we get from matrices the should run, so if it's not depended on something that depend on matrices it needs to have direct need.